gobblin-metrics-libs/gobblin-metrics/src/test/java/org/apache/gobblin/metrics/OpenTelemetryMetricsTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.metrics;
+
+import java.util.Map;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.configuration.State;
+
+
+public class OpenTelemetryMetricsTest  {
+
+  @Test
+  void testInitializeOpenTelemetryFailsWithoutEndpoint() {
+    State opentelemetryState = new State();
+    opentelemetryState.setProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENABLED, "true");
+    Assert.assertThrows(IllegalArgumentException.class, () -> {
+      OpenTelemetryMetrics.getInstance(opentelemetryState);
+    });
+  }
+
+  @Test
+  void testInitializeOpenTelemetrySucceedsWithEndpoint() {
+    State opentelemetryState = new State();
+    opentelemetryState.setProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENABLED, "true");
+    opentelemetryState.setProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENDPOINT, "http://localhost:4317");
+    OpenTelemetryMetrics metricsProvider = OpenTelemetryMetrics.getInstance(opentelemetryState);
+    System.out.println(metricsProvider.metricExporter.toString());
+  }
+
+  @Test
+  void testHeadersParseCorrectly() {
+    Map<String, String> headers = OpenTelemetryMetrics.parseHttpHeaders(
+        "{\"Content-Type\":\"application/x-protobuf\",\"headerTag\":\"tag1:value1,tag2:value2\"}");
+    Assert.assertEquals(headers.size(), 2);
+    Assert.assertEquals(headers.get("Content-Type"), "application/x-protobuf");
+    Assert.assertEquals(headers.get("headerTag"), "tag1:value1,tag2:value2");
```

Comment:
We don't check for ordering here, it's expecting the value of this key to be exactly the same order as its sent in which it should since it's just a string value